net/extras/genInventoryFile.py

```diff
@@ -54,6 +54,9 @@ def writeInventory(self, outFd):
             outFd.write(var_line)
             var_line = "netplugin_if={} ".format(connInfo[node]['data'])
             outFd.write(var_line)
+            role = connInfo[node].get('role', 'worker')
```

Comment:
@DivyaVavili, I have tested this and it works fine.  As per http://docs.ansible.com/ansible/playbooks_variables.html inventory_vars has higher precedence over host vars.